### PR TITLE
getEventById Test and Queries to match the expected value from the TifClientAPI

### PR DESCRIPTION
**Pull request by suryatiru**
- Minor name changes from "startTimestamp" to "startDateTime" and "endTimestamp" to "endDateTime"
- Modified the events.ts file to add the host to the attendeesList. Therefore, other test files need to be refactored to account for that. For example, instead of attendee we use "attendeesList[1]" since attendeesList[0] is the host 
```ts
  getEventById returns the data in the format as follows -> {
      id: string
      title: string
      description: string
      attendeeCount: number
      time: {
        secondsToStart: number
        timeZoneIdentifier: string | null
        dateRange: {
          startDateTime: Date
          endDateTime: Date
        }
         todayOrTomorrow: TodayOrTomorrow | null
      }
      previewAttendees: string[]
      location: {
        coordinate: {
          latitude: number,
          longitude: number
        },
        placemark: Omit<Placemark, "lat" | "lon"> | null,
        arrivalRadiusMeters: number
        isInArrivalTrackingPeriod: boolean
      }
      host: {
        relations: {
          themToYou: UserHostRelations
          youToThem: UserHostRelations
        }
        id: string
        username: string | null
        handle: string | null
        profileImageURL: string | null
      }
      settings: {
        shouldHideAfterStartDate: boolean
        isChatEnabled: boolean
      }
      chatExpirationTime: Date
      userAttendeeStatus: Role
      joinDate: Date | null
      hasArrived: boolean
      updatedAt: Date
      createdAt: Date
      endedAt: Date | null
    }
```

### CODE REVIEW

File: APILambda/events/getEventById.test.ts [hunk 1]
```diff
@@ -20,31 +29,94 @@ describe("GetSingleEvent tests", () => {
...
+    expect(resp.body).toEqual(
+      expect.arrayContaining([expect.objectContaining({
+        id: eventIds[0],
+        title: "Fake Event",
+        description: "",
+        attendeeCount: 2,
+        time: expect.objectContaining({
+          todayOrTomorrow: "Today"
+        }),
+        location: expect.objectContaining({
```
Comment (tifapp) [inline APILambda/events/getEventById.test.ts]: objectContaining not needed here, all of the exact values can be checked

File: APILambda/events/getEventById.test.ts [hunk 2]
```diff
@@ -20,31 +29,94 @@ describe("GetSingleEvent tests", () => {
...
+
+    const resp = await callGetEvent(token, eventIds[0])
+    expectTypeOf(resp.body).toMatchTypeOf<TiFEvent>()
+    expect(resp.body).toEqual(
+      expect.arrayContaining([expect.objectContaining({
+        id: eventIds[0],
+        title: "Fake Event",
+        description: "",
+        attendeeCount: 2,
+        time: expect.objectContaining({
```
Comment (tifapp) [inline APILambda/events/getEventById.test.ts]: objectContaining not needed here, all of the exact values can be checked

File: APILambda/events/getEventById.test.ts [hunk 3]
```diff
@@ -20,31 +29,94 @@ describe("GetSingleEvent tests", () => {
...
+        startDateTime: today.toDate(),
+        endDateTime: dayjs().add(1, "year").toDate()
       }
-    })
+    ], 1)
+
+    const resp = await callGetEvent(token, eventIds[0])
+    expectTypeOf(resp.body).toMatchTypeOf<TiFEvent>()
+    expect(resp.body).toEqual(
+      expect.arrayContaining([expect.objectContaining({
```
Comment (tifapp) [inline APILambda/events/getEventById.test.ts]: Dont return arrray, return first item from array

File: APILambda/events/getEventById.test.ts [hunk 4]
```diff
@@ -20,31 +29,94 @@ describe("GetSingleEvent tests", () => {
...
-        latitude: testEventInput.latitude,
-        longitude: testEventInput.longitude
+    addPlacemarkToDB(
+      conn,
+      {
+        lat: testEventInput.latitude,
+        lon: testEventInput.longitude,
+        name: "Sample Location",
+        city: "Sample Neighborhood",
+        country: "Sample Country",
```
Comment (tifapp) [inline APILambda/events/getEventById.test.ts:45]: Include country code in db

File: APILambda/events/getEventById.test.ts [hunk 5]
```diff
@@ -20,31 +29,94 @@ describe("GetSingleEvent tests", () => {
...
+            street: "Sample Street",
+            isoCountryCode: "",
+            streetNum: "1234"
+          }),
+          arrivalRadiusMeters: 120,
+          isInArrivalTrackingPeriod: true
+        }),
+        host: expect.objectContaining({
+          relations: expect.objectContaining({
+            themToYou: null,
```
Comment (tifapp) [inline APILambda/events/getEventById.test.ts]: should not be null, should be "not-friends"

File: APILambda/events/models.ts [hunk 6]
```diff
@@ -11,10 +11,20 @@ export const EventColorSchema = z.union([
...
-  title: string
+  title: string,
+  id: number,
+  host: {
+    username: string,
+    id: string,
+    handle: string,
+    profileImageURL?: string,
+    relations: {
+      themToYou: string | null,
```
Comment (suryatiru) [inline APILambda/events/models.ts:22]: Can't be null (default "not-friends")

File: APILambda/events/models.ts [hunk 7]
```diff
@@ -11,10 +11,20 @@ export const EventColorSchema = z.union([
...
-  handle: string
-  profileImageURL?: string,
-  title: string
+  title: string,
+  id: number,
+  host: {
+    username: string,
+    id: string,
+    handle: string,
+    profileImageURL?: string,
```
Comment (suryatiru) [inline APILambda/events/models.ts]: Add null

File: APILambda/events/models.ts [hunk 8]
```diff
@@ -11,10 +11,20 @@ export const EventColorSchema = z.union([
...
+    relations: {
+      themToYou: string | null,
+      youToThem: string | null
+    }
+  }
+  createdAt: Date,
+  updatedAt: Date | undefined
```
Comment (suryatiru) [inline APILambda/events/models.ts]: Replace undefined with null

File: APILambda/events/arrivals/getUpcomingEvents.ts [hunk 9]
```diff
@@ -10,20 +10,22 @@ type EventRegion = {
     longitude: number;
   };
   arrivalRadiusMeters: number;
-  isArrived: boolean;
+  hasArrived: boolean;
 }
 
 const mapEventsToRegions = (events: DatabaseEvent[]): EventRegion[] => {
   const eventRegions: Record<string, EventRegion> = {}
```
Comment (suryatiru) [inline APILambda/events/arrivals/getUpcomingEvents.ts:17]: Replace record with map for performance reasons

**Review by seanim0920 — APPROVED** [review]

**Pull request merged**